### ISSUE TITLE
pipes: stdin incorrectly triggers detection mode

### ISSUE THREAD

**Issue by ssbarnea** [ansible-community/ansible-lint#790]
Apparently current logic to trigger auto-detection mode is wrongly triggered when calling the linter using pipes:

```bash
cat test.yml | ansible-lint  # <-- this should lint that file alone
```

Assign it to me.

Note: apparently flake8 has the same behavior, I need to find-out if it is possible to distinguish reliably between being called without any arguments or it the file was fed directly to it.

The tricky part is that the `sys.stdin.isatty()` may not be a reliable way to do it as I suspect that some automation use-cases may return False.
